backend/howtheyvote/models/vote.py

```diff
@@ -78,6 +79,127 @@ class ProcedureStage(Enum):
     OLP_THIRD_READING = "OLP_THIRD_READING"
 
 
+class AmendmentAuthorType(Enum):
+    GROUP = "GROUP"
+    COMMITTEE = "COMMITTEE"
+    ORIGINAL_TEXT = "ORIGINAL_TEXT"
+    MEMBERS = "MEMBERS"
+    ORALLY = "ORALLY"
+    RAPPORTEUR = "RAPPORTEUR"
+
+
+@dataclasses.dataclass
+class AmendmentAuthorGroup:
+    group: Group | None
+
+
+@dataclasses.dataclass
+class AmendmentAuthorCommittee:
+    committee: Committee | None
+
+
+@dataclasses.dataclass
+class AmendmentAuthorOriginalText:
+    pass
+
+
+@dataclasses.dataclass
+class AmendmentAuthorMembers:
+    pass
+
+
+@dataclasses.dataclass
+class AmendmentAuthorOrally:
+    pass
+
+
+@dataclasses.dataclass
+class AmendmentAuthorRapporteur:
+    pass
+
+
+AmendmentAuthor = (
+    AmendmentAuthorGroup
+    | AmendmentAuthorCommittee
+    | AmendmentAuthorOriginalText
+    | AmendmentAuthorMembers
+    | AmendmentAuthorOrally
+    | AmendmentAuthorRapporteur
+)
+
+AMENDMENT_AUTHOR_TYPE_TO_CLASS: dict[AmendmentAuthorType, type[AmendmentAuthor]] = {
```

Comment:
This way of modeling amendment authors is quite verbose, but I didn’t find a simpler way that’s still type safe, i.e. I want this to work without manual type casting or type errors:

```py
for author in vote.amendment_authors:
  if author.type == AmendmentAuthorType.GROUP:
    print(author.group.label)
  # ...
```